app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt

```diff
@@ -0,0 +1,155 @@
+    private fun onSuccessGetLobbies(response: Response) {
+        if (response.isSuccessful) {
+            val responseBody = response.body?.string()
+            if (responseBody != null) {
+                val openLobbies = Array(JSONArray(responseBody).length()) {
+                    JSONArray(responseBody)[it]
+                }
+                for ((index, lobby) in openLobbies.withIndex()) {
+                    Log.d("LO", lobby.toString())
+                    val jsonLobby = JSONObject(lobby.toString())
+                    lobbies.add(
+                        index, JoinLobbyLobby(
+                            jsonLobby.getString("name"),
+                            jsonLobby.getJSONArray("players").length(),
+                            jsonLobby.getInt("maxPlayers"),
+                            jsonLobby.getString("lobbyid")
+                        )
+                    )
+                }
+            }
+            runOnUiThread {
+                adapter.notifyDataSetChanged()
+            }
+
+            Log.d("LO", lobbies.toString())
+        }
+    }
+
+
+    fun btnJoinCode(view: View) {
+        jLobbyWithCode(null)
+    }
+
+    private fun jLobbyWithCode(response: Response?) {
+        if (lobbyToJoin.lobbyID != "") {
+            joinLobby(lobbyToJoin)
+            return
+        }
+        val builder: AlertDialog.Builder = AlertDialog.Builder(this)
+        builder.setTitle("Enter LobbyID")
+        val input = EditText(this)
+
+        input.setInputType(InputType.TYPE_CLASS_TEXT)
+        builder.setView(input)
+        builder.setPositiveButton(
+            "OK"
+        ) { _, _ ->
+            lobbyToJoin = LobbyJoin(input.getText().toString())
+            joinLobby(lobbyToJoin)
+        }
+        builder.setNegativeButton(
+            "Cancel"
+        ) { dialog, _ -> dialog.cancel() }
+        builder.show()
+    }
+
+    private fun joinLobby(lobbyToJoin: LobbyJoin) {
+        HttpClient.post(
+            "/lobbys/join",
+            Gson().toJson(lobbyToJoin),
+            StoreToken(this).getAccessToken(),
+            CallbackCreator().createCallback(::onFailureLobby, ::onSuccessJoinLobby)
+        )
+    }
+
+    private fun onFailureLobby() {
+        this.runOnUiThread {
+            Toast.makeText(this, "Lobby functionality failed", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun onSuccessJoinLobby(response: Response) {
+        Log.d("Lobby", response.toString())
+
+        if (response.isSuccessful) {
+            response.body?.string()?.let {
+                Log.d("Lobby", it)
+                val intent = Intent(this, LobbyActivity::class.java)
+                intent.putExtra("lobbyCode", it)
+                startActivity(intent)
+            }
+        } else {
+            HttpClient.get(
+                "/lobbys/leave",
+                StoreToken(this).getAccessToken(),
+                CallbackCreator().createCallback(::onFailureLobby, ::jLobbyWithCode)
```

Comment:
This could also produce an infinite loop if there is an error unrelated to the player being in a lobby

Reply:
Will be fixed in #21